### PR TITLE
perf(evt): let the evt/pet chunk size be configured, and raise it

### PR DESCRIPTION
**Pull request by ggmarshall**
Stacked on #161 (`fable-updates`), because it edits `scripts/tier/evt.py` as refactored there. Retarget to `main` once #161 lands. The diff here is one commit.

Companion to legend-exp/pygama#710 — that PR does the heavy lifting; this one turns the remaining knob.

## What and why

`build_evt` reads the TCM in chunks and pays a fixed cost per chunk, per operation, per channel. That cost does not shrink with the chunk size, so the chunk **count** is what drives the runtime.

A cal file holds ~1.08 M events against ~5 k for a phy file. At pygama's default `buffer_len=10**4` a cal file was split into **108 chunks and paid the overhead 108 times, while phy paid it once** — which is the whole reason evt is disproportionately slow for calibration data. Nothing in the dataflow ever set the value.

This adds `--buffer-len` to `build-tier-evt`, resolved as **CLI > the `tier_evt` rule config > the pygama default**, and passes it to *both* `build_evt` calls — the main one and the muon pass, which was also silently running at the default. The rules take it from a new `evt_buffer_len` key in `dataflow-config.yaml`, set to `10**6` so a cal file is a single chunk.

## Effect

On a production p16 r000 cal `pet` build, together with the pygama caching work:

| | wall clock | peak RSS |
|---|---|---|
| before | **44 m 00 s** | 1.43 GB |
| pygama#710 only, same chunking | 12 m 22 s | 1.45 GB |
| + this change | **1 m 34 s** | 1.63 GB |

Memory is not a concern: because `build_evt` is called without an output file it already held every chunk in memory, so fewer, larger chunks is not a regression — measured peak RSS barely moves.

## Safety

Raising this is only sound now that `first_at`/`last_at` no longer depend on where the chunk boundaries fall — that is the fix in legend-exp/pygama#710, and **this PR should not be merged before it**. With the old aggregator, changing `buffer_len` silently changed `trigger___timestamp`.

Verified end to end against the production `pet` files for both cal and phy: every dataset, dtype and attribute compared.

## Not run

`./tests/runprod/run-all.sh` — it needs `install.sh` to clone legend-metadata into `inputs/`, which my working tree does not have. The DAG is unchanged by construction: `evt.smk` gains only a `params:` entry and one token in the shell string, with no change to any rule's inputs, outputs or wildcards. Worth letting CI confirm.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: workflow/src/legenddataflow/scripts/tier/evt.py [hunk 1]
```diff
@@ -86,6 +97,13 @@ def build_tier_evt() -> None:
     chmap = LegendMetadata(args.metadata, lazy=True).channelmap(on=args.timestamp)
     evt_config = AttrsDict(Props.read_from(df_config.inputs.evt_config))
 
+    # CLI wins over the rule config, which wins over the pygama default
+    buffer_len = args.buffer_len
+    if buffer_len is None:
+        buffer_len = df_config.get("options", {}).get("buffer_len", 10**4)
+    buffer_len = int(buffer_len)
+    log.debug("using buffer_len=%d", buffer_len)
```
[the remark this answers is not in the document]
Comment (ggmarshall) [reply workflow/src/legenddataflow/scripts/tier/evt.py]: Fixed in bb335d6. Resolution and validation now live in one testable helper, `_resolve_buffer_len`: it rejects non-positive and non-integer values with a message naming which of the two sources (`--buffer-len` or the tier_evt rule config) the bad value came from, so the failure points at the setting rather than surfacing later inside the read loop.

One judgement call while adding tests: an explicit `buffer_len: null` now means "unset" and falls through to the default, rather than erroring as a non-integer. That matches how `null` is already used for `muon_config` in these rule configs.

**Comment by ggmarshall** [comment]
Merged the updated base (`c2c5d11`) — **CI is green now (12/12)**.

The unit-test failures here weren't from this PR's change: they were `ImportError: cannot import name 'check_input_files' from 'legenddataflowscripts.utils'`, i.e. the stale `legend-dataflow-scripts==0.3.3` pin, which predates the helpers #161 imports. Picking up #161's merge of main brings `0.4.0a1` (plus `pygama==2.7.0a1`, `dbetto>=1.4`) and the imports resolve.

No changes to the buffer-len work itself.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
